runtime-components/non-repudiation/src/main/scala/com/daml/nonrepudiation/MetricsReporterOwner.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation
+
+import com.codahale.metrics.{MetricRegistry, ScheduledReporter, Slf4jReporter}
+import com.daml.resources._
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+// We don't need access to the underlying resource, we use
+// the owner only to manage the reporter's life cycle
+sealed abstract class MetricsReporterOwner[Context: HasExecutionContext]
+    extends AbstractResourceOwner[Context, Unit]
+
+object MetricsReporterOwner {
+
+  def noop[Context: HasExecutionContext](registry: MetricRegistry): MetricsReporterOwner[Context] =
```

Comment:
instead of having a noop owner, how about making the field optional and just setting it to `None` where we don’t care?

Reply:
This raises a fair point, which is that the absence of metrics reporting is mostly an edge case for testing. I don't think an `Option` makes too much sense due to this, but I'll have a second look and see if there's a cleaner approach. Thanks! 🙇🏻

Reply:
Addressed in https://github.com/digital-asset/daml/pull/8766/commits/fb25213e71e185acdafb40da84f063853519a9c0

1. It didn't really made sense to initialize and manage the metrics registry from within the proxy owner: the database will need to be instrumented as well and those are external dependencies as well, so it makes sense for the registry to be managed externally
2. A singleton, although probably not extremely clean, is good enough for now (and the reliance on a singleton is fairly consistent with how we deal with loggers)
3. The owner right now is simply managed externally, so if you to start one you can do it from whatever runs the proxy, the proxy will simply rely on dropping metrics in the right registry
4. Where we are not interested about metrics reporting, the entire thing simply disappears, which is nice

I don't like 100% the singleton, but I have to say that the approach seems to make sense for metrics and it seems to make everything a lot simpler.